datafusion/sql/src/unparser/plan.rs

```diff
@@ -708,6 +726,38 @@ impl Unparser<'_> {
         }
     }
 
+    /// Try to find the placeholder column name generated by `RecursiveUnnestRewriter`
+    /// Only match the pattern `Expr::Alias(Expr::Column("unnest_placeholder(...)"))`
+    fn is_unnest_placeholder(expr: &Expr) -> bool {
+        if let Expr::Alias(Alias { expr, .. }) = expr {
+            if let Expr::Column(Column { name, .. }) = expr.as_ref() {
+                return name.starts_with("unnest_placeholder");
```

Comment:
is this a well known constant name? should it be defined somewhere?

also, internally create names should be something a user is very unlikely to have in their data
perhaps starting with `$` or at least `_`

Reply:
Sounds great. I follow the style of `__scalar_sq` and `__correlated_sq` to name it `__unnest_placeholder`.